azure/functions_worker/bindings/meta.py

```diff
@@ -173,14 +173,14 @@ def _parse_datetime(
         ]
         dt = None
 
-        too_fractional = re.match(r'.*\.\d{6}(\d+)Z', datetime_str)
+        too_fractional = re.match(r'.*\.\d{6}(\d+)', datetime_str)
```

Comment:
This would make inputs that *do* send `Z` fail, so you want `Z?` here instead.

Reply:
It had Z and was failing because of it, as the manual trigger is 9999-12-31T23:59:59.9999999+00:00 with no Z.  This should work with the Z as well as it should truncate from before the Z.  I can double check.

Reply:
Yes, in my test harness it will move the Z correctly.  Technically it strips it off then re-adds the Z on line 183

Reply:
Notice on 183 it will pull from the start of the string all the way until the match, so it is short of the Z, but adds one there.

Reply:
Right, in this case changing `Z` to `(Z|[\+|-]\d{1,2}:\d{1,2})` is more appropriate.  We don't want so silently truncate the timezone information.

Reply:
Updated with code to grab the TZ and append it if it was provided.  Also, updated the format filters to include - (negative) TZ offsets.

Reply:
It will still add Z if that is the format given.